hazelcast-client/src/main/java/com/hazelcast/client/impl/HazelcastClientInstanceImpl.java

```diff
@@ -242,6 +244,8 @@ public HazelcastClientInstanceImpl(ClientConfig config,
         lockReferenceIdGenerator = new ClientLockReferenceIdGenerator();
         nearCacheManager = clientExtension.createNearCacheManager();
         clientExceptionFactory = initClientExceptionFactory();
+
+        statistics = new Statistics(this);
```

Comment:
@pveentjer I tried not using "this" but I end up needing it due to this line: https://github.com/ihsandemir/hazelcast/blob/clientStats/hazelcast-client/src/main/java/com/hazelcast/client/impl/statistics/Statistics.java#L196  ClientInvocation constructor needs ClientInstance, hence it seems like I can not get rid of it unless ClientInvocation is also changed. We can open a cleanup issue for this since this change affects a number of different places ("this" is used a lot in HazelcastClientInstanceImpl).